src/main/java/com/ppojin/util/StringAddCalculator.java

```diff
@@ -0,0 +1,53 @@
+package com.ppojin.util;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final String DEFAULT_DELIMITER = ",|:";
+    private static final String CUSTOM_DELIMITER_CONVENTION = "^//(.)\n(.*)$";
+    private static final int CUSTOM_DELIMITER_CONVENTION_END_INDEX = 4;
+
+    private StringAddCalculator() {
+    }
+
+    public static int splitAndSum(String text) {
+        if (StringUtils.isEmpty(text)) {
+            return 0;
+        }
+
+        final String delimiter = getDelimiter(text);
+
+        if (isCustomDelimiter(delimiter)) {
+            text = text.substring(CUSTOM_DELIMITER_CONVENTION_END_INDEX);
+        }
+
+        final String[] nums = StringUtils.splitByDelimiter(text, delimiter);
+
+        return Arrays.stream(nums)
+                .map(StringAddCalculator::parseIntExceptNegative)
+                .mapToInt(i -> i)
+                .sum();
+    }
+
+    private static boolean isCustomDelimiter(String delimiter) {
+        return !delimiter.equals(DEFAULT_DELIMITER);
+    }
+
+    private static String getDelimiter(String text) {
+        Matcher m = Pattern.compile(CUSTOM_DELIMITER_CONVENTION).matcher(text);
```

Comment:
계산기를 호출할 때 마다 정규식을 매번 컴파일하겠네요.
정규식 비용은 비싸기 때문에 고정된 값이라면 패턴 자체를 상수화시키는 것이 좋습니다.

<img width="585" alt="image" src="https://github.com/g-oop/goop-java-racingcar/assets/36617783/aac31e95-968e-4466-9f25-a1f7846f932a">

```suggestion
        Matcher matcher = PATTERN.matcher(text);
```